src/lib/utils.jl

```diff
@@ -99,6 +99,26 @@ macro showgrad(x)
     end)
 end
 
+"""
+jacobian(f,x)
+
+Construct the Jacobian of `f` where `x` is a real-valued array
+and `f(x)` is also a real-valued array.
+"""
+function jacobian(f,x)
+    y,back  = Zygote.pullback(f,x)
+    k  = length(y)
+    n  = length(x)
+    J  = Matrix{eltype(y)}(undef,k,n)
+    e_i = fill!(similar(y), 0)
+    @inbounds for i = 1:k
+        e_i[i] = oneunit(eltype(x))
+        J[i,:] = back(e_i)[1]
+        e_i[i] = zero(eltype(x))
+    end
+    J
```

Comment:
```suggestion
    (J,)
```
Returning a 1-tuple allow for a consistent `jacobian(f, x, y)` without breaking anything, and would match what `gradient` does.